connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/WorkerCoordinator.java

```diff
@@ -317,12 +317,14 @@ public double measure(MetricConfig config, long now) {
                 }
             };
 
-            metrics.addMetric(metrics.metricName("assigned-connectors",
-                              this.metricGrpName,
-                              "The number of connector instances currently assigned to this consumer"), numConnectors);
-            metrics.addMetric(metrics.metricName("assigned-tasks",
-                              this.metricGrpName,
-                              "The number of tasks currently assigned to this consumer"), numTasks);
+            if (assignmentSnapshot != null) {
```

Comment:
With this change, if `assignmentSnapshot` is null when we enter this constructor, the metrics will never get registered, even if it gets initialized shortly after.
Should we still register the metrics, and return `Double.NaN` in both `measure()` method if `assignmentSnapshot` is null?

Reply:
@mimaison that sounds a good option that I have also considered. Let me make this change :)